src/github/branch.ts

```diff
@@ -29,19 +26,19 @@ export const createBranch = async (
 		});
 	} else {
 
-		const jiraPayload = await transformBranch(context.github, webhookPayload);
+		const gitHubClient = new GitHubClient(getCloudInstallationId(githubInstallationId), context.log);
+		const github = await booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_BRANCH_EVENT, false, jiraClient.baseURL) ? gitHubClient : context.github;
+		const jiraPayload = await transformBranch(github, webhookPayload);
 
 		if (!jiraPayload) {
 			context.log(
-				{ noop: "no_jira_payload_create_branch" },
+				{ noop: "no_jira_payload_create_branch" }, 
```

Comment:
/nit `noop` data is from oldschool code that needs to be removed.  If you're editing anything else in this PR, please remove it as it adds no value.